src/web/components/ContributionSlot.tsx

```diff
@@ -0,0 +1,20 @@
+import { css } from '@emotion/react';
+
+export const ContributionSlot = () => {
+	return (
+		<div
+			css={css`
+				position: sticky;
+				top: 0;
+			`}
+		>
+			<a href="https://support.theguardian.com/uk/subscribe/digital?acquisitionData=%7B%22componentType%22%3A%22ACQUISITIONS_OTHER%22%2C%22source%22%3A%22GUARDIAN_WEB%22%2C%22campaignCode%22%3A%22shady_pie_open_2019%22%2C%22componentId%22%3A%22shady_pie_open_2019%22%7D&INTCMP=shady_pie_open_2019">
```

Comment:
It might be worth using [`URLSearchParams`][mdn] here to make this more readable?

[mdn]: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
```js
const params = new URLSearchParams();
params.set(
	'acquisitionData',
	JSON.stringify({
		componentType: 'ACQUISITIONS_OTHER',
		source: 'GUARDIAN_WEB',
		campaignCode: 'shady_pie_open_2019',
		componentId: 'shady_pie_open_2019',
	}),
);
params.set('INTCMP', 'shady_pie_open_2019');

// you can then use:

params.toString()
```

Reply:
Fixed upstream: https://github.com/guardian/frontend/pull/24050/commits/9dc5851e17fd792f9e825ba888fb7df6d4c5899e

Reply:
Without a doubt, that code is easier to read and understand the url. But it's harder to verify that the url works and maybe harder to change it? When it's simply a string I can just hold down the CMD key on my laptop and click it and, yep, that works...

Something else to consider is if we keep it raw then we make it possible for non devs to change this code - something I think we should invite more of in general but is perhaps useful in this context?

To be honest I'm not sure, `URLSearchParams ` is a far superior way for managing urls... 🤷 this really depends on how this code will be used going forwards.

Reply:
Another note on `URLSearchParams` is that it's [not supported in internet explorer](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#browser_compatibility)

Reply:
If you click this URL in your IDE, you’ll create a fake event, so not sure it’s really an issue anyways? This gets expanded on the page, too.

You can even share a [Typescript Play][ts] example for people who don’t want to open the console.

[ts]: https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgQwE4ILYRgXhmApgdxgFUAlAGQGVdlgALABWTQgAoBKAbgFgAoRFdADoIuKC14BIAOQJgARwCuASwhKoS8ABEEUBFIA0kgFIUA8gDlhUJErABzJQDMAniwDekiaFRxwuMFAAKs5wuABcMFIAggDCAIpEAJIUiYGJFhQA+qaBABIAoiQGnhAgCkjA4ZEA4kRRJJqJUeaZAOr5AELFPBJeaIhKdmAxIAAmVVIQtAijzplwSriZIKFgmQBMAAwAjACc3b3evngBiaMRk9Oz84vLqxs7+4Y9AL5sz5y8-MzCoixSiXMgRiAFl6AZIlMZnMFksVv4HnspJ8eLwAPRomDOMowYAIMAwKC0fwwBQiMK8XigSAgAA2uEEtJAdhY3yEUBAFGsthZbDYQA

Reply:
> Another note on `URLSearchParams` is that it's [not supported in internet explorer](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#browser_compatibility)

Are there Ad Blockers in Internet Explorer?

Reply:
> Are there Ad Blockers in Internet Explorer? Can we [polyfill](https://github.com/ungap/url-search-params) it?

Believe there are a few like AdblockPlus though catering for IE is probably an edge case here

Reply:
> catering for IE is probably an edge case here

I once suggested we think about not supporting the top right slot on IE because, well, it's just IE... I was surprised to learn that this one little slot just on IE makes quite a lot of money each month!

Reply:
Is this not rendered server-side anyways? If not, surely we could!

Reply:
> Is this not rendered server-side anyways?

It has to be client side because the ad detection script does some dom manipulation to decide if the browser is using an ad blocker or not.

We _could_ generate the url server side and pass it down via `window` but this would add complexity which we probably don't want?

Reply:
I checked and polyfill.io has it: https://polyfill.io/v3/polyfill.min.js?features=URLSearchParams

Is this enough to carry on and just implement this in a readable manner?

Reply:
I've made a commit so that this url is now constructed using `URLSearchParams`. I'm still generally more in favour of simpler code but I also see the benefit to this approach so I'm happy to follow the Reviewer Wins philosophy here 😺